### ISSUE TITLE
Printing via for loops broken by kable_styling

### ISSUE THREAD

**Issue by jnaber** [haozhu233/kableExtra#532]
When I try to use a for loop to iteratively add several tables to an html document, the use of kable_styling causes the output to show up as several html files, each with a single table, not one file with several tables inside. I'm on R 4.0.2 with kableExtra 1.2.1. 

For example, if I knit the following to HTML, I get three new windows, each with one table of data, and a markdown document that has no data inside:

>\```{r setup, include=FALSE}
>knitr::opts_chunk$set(echo = TRUE, results = "asis")
>\```

> \```{r mtcars}
> library(tidyverse)
> library(kableExtra)
> for(i in c(4, 6, 8)) {
>   mtcars %>%
>     filter(cyl == i) %>%
>     kable() %>%
>     kable_styling() %>%
>     print()
>   cat("\n")
> }
> \```

It appears the issue does require both a loop and the kable_styling function. If I break apart the loop into three chunks of code, one for 4, 6, and 8 respectively, then I get the single document with the data inside as expected. Omitting the kable_styling() line results in a single markdown document, with the data inside as expected.  It's only together the issue shows up.

I did find an old issue, [#160](https://github.com/haozhu233/kableExtra/issues/160), where there was a similar problem with the output and pipes were the issue. I've found that using them or not doesn't affect the output here.

**Comment by vincentarelbundock** [comment]
Have you tried with `cat` instead of `print` and with `results="asis"` in the chunk header?

````
---
output: html_document
---

```{r, message=FALSE, results="asis"}
library(dplyr)
library(kableExtra)

for (i in c(4, 6, 8)) {
	mtcars %>%
		filter(cyl == i) %>%
		kbl() %>%
		kable_styling() %>%
		cat
}
```
````

**Comment by haozhu233** [comment]
Just want to confirm that with current fix, we went back to the old behavior in 1.1.0 where you can use the print statement to do the tricks.

**Comment by jnaber** [comment]
Thank you for you help with this

Get Outlook for Android<https://aka.ms/ghei36>

________________________________
